src/main/kotlin/it/pagopa/wallet/services/WalletService.kt

```diff
@@ -1315,6 +1328,45 @@ class WalletService(
             }
     }
 
+    private fun buildOutcomeUrl(wallet: Wallet): Mono<Pair<URI, URI>> {
+        val isTransactionWithContextualOnboard =
+            wallet
+                .getApplicationMetadata(
+                    pagopaWalletApplicationId,
+                    WalletApplicationMetadata.Metadata.PAYMENT_WITH_CONTEXTUAL_ONBOARD)
+                .toBoolean()
+        return if (isTransactionWithContextualOnboard) {
+            walletJwtTokenCtxOnboardingTemplateWrapper.findById(wallet.id.value.toString()).map {
+                session ->
+                Pair(
+                    UriComponentsBuilder.fromUriString(
+                            sessionUrlConfig.trxWithContextualOnboardingBasePath.plus(
+                                sessionUrlConfig
+                                    .trxWithContextualOnboardingOutcomeSuffix)) // append query
+                        // param to
+                        // prevent
+                        // caching
+                        .queryParam("t", Instant.now().toEpochMilli())
+                        .build(mapOf("sessionToken" to session.jwtToken)),
```

Comment:
the configured urls does not have a {sessionToken} part to be replaced (see helm values). moreover we have to set also transactionId and clientId, all as fragment parameters. right?